crates/transpiler/src/passes/constrained_reschedule.rs

```diff
@@ -0,0 +1,304 @@
+/// * `clbit_write_latency` - Additional latency for classical bit write operations
+/// * `pulse_align` - Alignment constraint for gate operations (in dt units)
+/// * `acquire_align` - Alignment constraint for measurement/reset operations (in dt units)
+/// * `target` - Optional target backend for duration information
+fn push_node_back(
+    dag: &DAGCircuit,
+    node_index: NodeIndex,
+    node_start_time: &Bound<PyDict>,
+    clbit_write_latency: u64,
+    pulse_align: u64,
+    acquire_align: u64,
+    target: Option<&Target>,
+) -> PyResult<()> {
+    let op = match dag.dag().node_weight(node_index) {
+        Some(NodeType::Operation(op)) => op,
+        _ => panic!("topological_op_nodes() should only return instances of DagOpNode."),
+    };
+
+    let op_view = op.op.view();
+    let alignment = match op_view {
+        OperationRef::Gate(_) | OperationRef::StandardGate(_) => Some(pulse_align),
+        OperationRef::StandardInstruction(StandardInstruction::Reset)
+        | OperationRef::StandardInstruction(StandardInstruction::Measure) => Some(acquire_align),
+        OperationRef::StandardInstruction(StandardInstruction::Delay(_)) => None,
+        _ => None,
+    };
+
+    let obj = node_start_time.get_item(node_index.index())?;
+    let mut this_t0: f64 = obj
+        .as_ref()
+        .ok_or_else(|| PyValueError::new_err("Missing value in node_start_time"))?
+        .extract()?;
+
+    if let Some(alignment) = alignment {
+        let misalignment = this_t0 % alignment as f64;
+        let shift = if misalignment != 0.0 {
+            (alignment as f64 - misalignment).max(0.0)
+        } else {
+            0.0
+        };
+        this_t0 += shift;
+        node_start_time
+            .set_item(node_index.index(), this_t0)
+            .unwrap();
+    }
+
+    let new_t1q = if let Some(target) = target {
+        let qargs: Vec<PhysicalQubit> = dag
+            .qargs_interner()
+            .get(op.qubits)
+            .iter()
+            .map(|q| PhysicalQubit(q.index() as u32))
+            .collect();
+        let duration = target.get_duration(op.op.name(), &qargs).unwrap_or(0.0);
+        this_t0 + duration
+    } else if matches!(
+        op_view,
+        OperationRef::StandardInstruction(StandardInstruction::Delay(_))
+    ) {
+        let duration = if let Some(param) = op.params_view().first() {
+            match param {
+                Param::Obj(val) => {
+                    // Try to extract as different numeric types
+                    val.bind(node_start_time.py())
+                        .extract::<f64>()
+                        .or_else(|_| {
+                            val.bind(node_start_time.py())
+                                .extract::<u32>()
+                                .map(|v| v as f64)
+                        })
+                        .unwrap_or(0.0)
+                }
+                Param::Float(f) => *f,
+                _ => 0.0,
```

Comment:
Is this correct? If it's not a float or an obj then it's a parameter expression which I believe is supported (for better or worse), that doesn't mean the duration is zero does it? Looking at the python code I don't think anything would fail, it wouldn't make any sense and returning an error might make more sense to handle a parameterized delay.